update/sync/syncAccountsDBs.go

```diff
@@ -0,0 +1,210 @@
+package sync
+
+import (
+	"bytes"
+	"sync"
+	"time"
+
+	"github.com/ElrondNetwork/elrond-go/core"
+	"github.com/ElrondNetwork/elrond-go/core/check"
+	"github.com/ElrondNetwork/elrond-go/data"
+	"github.com/ElrondNetwork/elrond-go/data/block"
+	"github.com/ElrondNetwork/elrond-go/data/state"
+	"github.com/ElrondNetwork/elrond-go/update"
+	"github.com/ElrondNetwork/elrond-go/update/genesis"
+)
+
+type syncAccountsDBs struct {
+	tries              *concurrentTriesMap
+	accountsBDsSyncers update.AccountsDBSyncContainer
+	activeAccountsDBs  map[state.AccountsDbIdentifier]state.AccountsAdapter
+	mutSynced          sync.Mutex
+	synced             bool
+}
+
+// ArgsNewSyncAccountsDBsHandler is the argument structured to create a sync tries handler
+type ArgsNewSyncAccountsDBsHandler struct {
+	AccountsDBsSyncers update.AccountsDBSyncContainer
+	ActiveAccountsDBs  map[state.AccountsDbIdentifier]state.AccountsAdapter
+}
+
+// NewSyncAccountsDBsHandler creates a new syncAccountsDBs
+func NewSyncAccountsDBsHandler(args ArgsNewSyncAccountsDBsHandler) (*syncAccountsDBs, error) {
+	if check.IfNil(args.AccountsDBsSyncers) {
+		return nil, update.ErrNilAccountsDBSyncContainer
+	}
+
+	st := &syncAccountsDBs{
+		tries:              newConcurrentTriesMap(),
+		accountsBDsSyncers: args.AccountsDBsSyncers,
+		activeAccountsDBs:  make(map[state.AccountsDbIdentifier]state.AccountsAdapter),
+		synced:             false,
+		mutSynced:          sync.Mutex{},
+	}
+	for key, value := range args.ActiveAccountsDBs {
+		st.activeAccountsDBs[key] = value
+	}
+
+	return st, nil
+}
+
+// SyncAccountsDBsFrom syncs all the state tries from an epoch start metachain
+func (st *syncAccountsDBs) SyncTriesFrom(meta *block.MetaBlock, waitTime time.Duration) error {
```

Comment:
Adding a maximum wait time is a good practice not to have go routines that will never end but the value will be very hard to adjust or preset. Suggestion to use a context.Context instead.

Reply:
next PR